### PR TITLE
chore: A base scrubber and a "Shared Access Signature" Scrubber

### PR DESCRIPTION
**Pull request by saileshbaidya**
## Related Issues/PRs

Task: 2363740
Close #2363740

#xxx

## What changes are proposed in this pull request?

Our telemetry system has to avoid logging prohibited information like Shared Access Token, Secrets, etc. 
This PR includes a foundational base class for Scrubbers. I am also adding a concrete scrubber SASScrubber to address the latest reported SAS token leaks.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [*] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [*] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala [hunk 1]
```diff
@@ -40,6 +42,11 @@ object SynapseMLLogging extends Logging {
     logInfo(s"metrics/ ${mapToPrint.toJson.compactPrint}")
   }
 
+  def logMessage(message: String): Unit = {
+    val scrubbedMessage = SASScrubber.scrub(message)
+    logInfo(s"sending $scrubbedMessage")
```
Comment (mhamilton723) [inline core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala:47]:
```suggestion
    logInfo(SASScrubber.scrub(message))
```

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
Nicely done! Minor nits only

File: core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala (hunk 1 above)
Comment (mhamilton723) [reply core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala:47]: no need to have sending in this more general API

Comment (saileshbaidya) [reply core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala:47]: Great. Thanks for the suggestion. That append of a prefix "sending" slipped in unintentionally.

**Review by mhamilton723 — DISMISSED** [review]

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**